examples/pmatch.kl

```diff
@@ -5,45 +5,29 @@
 
 (import (shift "prelude.kl" 1))
 
-(import (shift "quasiquote.kl" 1))
+(import (shift "identifier.kl" 1))
+(import (shift "dot-dot-dot.kl" 1))
 (import (shift "defun.kl" 1))
 (import (shift "lispy-do.kl" 1))
 (import (shift "free-identifier-case.kl" 1))
 (import (shift "temporaries.kl" 1))
 (import (shift "pair-datatype.kl" 1))
 (import (shift "list-datatype.kl" 1))
 
-(meta
-  (define identifier?
-    (lambda (stx)
-      (syntax-case stx [(ident x) (true)] [_ (false)]))))
-
 (meta
   (datatype (Maybe A)
     (nothing)
     (just A)))
 
 (meta
   (defun syntax->list (stx)
-    (syntax-case stx
-      [() (nil)]
-      [(cons x xs) (:: x (syntax->list xs))])))
-
-(meta
-  (define else-case
-     (lambda (stx)
-       (syntax-case stx
-         [(list (e x))
-          (if (identifier? e)
-              (free-identifier-case e
-                [else (pure (just x))]
-                [(else other) (pure (nothing))])
-              (pure (nothing)))]
-         [(ident e)
-          (free-identifier-case e
-            [else (pure (just 'else-x))]
-            [(else other) (pure (nothing))])]
-         [_ (pure (nothing))]))))
+    (syntax-case stx ()
+      [()
+       (pure (nil))]
+      [(,x ,xs ...)
+       (>>= (syntax->list xs)
+         (lambda (list)
+           (pure (:: x list))))])))
```

Comment:
This definition of `syntax->list` is longer than before... and it has a different type! That's because while `raw-syntax-case` is a pure expression, `fancy-syntax-case` runs in the `Macro` monad, so that we can use `free-identifier=?` to compare identifiers to keywords.